File: backend/NPVCalculator/src/Npv.Contracts/Responses/NpvCalculationResponse.cs
using Npv.Contracts.Models;
using Npv.Contracts.Requests;

namespace Npv.Contracts.Responses;

public class NpvCalculationResponse
{
    public string CalculationId { get; set; } = string.Empty;
    public List<NpvResult> Results { get; set; } = new();
    public string Status { get; set; } = "Completed";
    public DateTime CalculatedAt { get; set; } = DateTime.UtcNow;
    public NpvRequest Request { get; set; } = new NpvRequest();
}

File: frontend/npv-calculator/src/app/core/models/npv-calculation-response.ts
import {NpvResult} from './npv-result';
import {NpvRequest} from './npv-request';

export interface NpvCalculationResponse {
  calculationId: string;
  results: NpvResult[];
  status: string;
  calculatedAt: string;
  request: NpvRequest;
}
